File: src/main/java/com/mmust/token/SubtractOpToken.java
package com.mmust.token;

public class SubtractOpToken extends OperatorToken {

	public SubtractOpToken() {
		super("-");
	}

	@Override
	public double eval(double op1, double op2) {
		return op1 - op2;
	}

	@Override
	public int getPrecedence() {
		return 2;
	}

	@Override
	public Associativity getAssociativity() {
		return Associativity.LEFT;
	}
}

File: src/main/kotlin/com/mmust/token/SubtractOpToken.kt
package com.mmust.token

class SubtractOpToken : OperatorToken("-") {

    override val precedence: Int
        get() = 2

    override val associativity: Associativity
        get() = Associativity.LEFT

    override fun eval(op1: Double, op2: Double): Double {
        return op1 - op2
    }
}
